infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -110,6 +118,13 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                     logEntrySyncState.setTransitionEventId(event.getEventId());
                     fsm.setBaseSnapshot(event.getMetadata().getLastTransferredBaseSnapshot());
                     fsm.setAckedTimestamp(event.getMetadata().getLastLogEntrySyncedTimestamp());
+                    if (tableManagerPlugin.isUpgraded()) {
+                        // If LR is in upgrading path, it means this cycle of snapshot sync was triggered
+                        // forcibly because LR detected a version mismatch. Flipping the flag back to false
+                        // here to indicate that the upgrade path is completed.
+                        log.info("Forced snapshot sync triggered by upgrading has finished. Flip the upgrade flag back to false");
```

Comment:
This message is too long right? If we are too verbose everywhere we contribute to log size (with non-relevant info). Perhaps: "Forced snapshot sync due to LR upgrade COMPLETED." (and once you find this message you know the flag was flipped, anyways you are not printing the actual value of the flag).